src/main/java/htsjdk/samtools/util/IOUtil.java

```diff
@@ -246,6 +246,15 @@ public static boolean isRegularPath(final File file) {
     }
 
     /**
+     * @return true if the path is not a device (e.g. /dev/null or /dev/stdin), and is not
+     * an existing directory.  I.e. is is a regular path that may correspond to an existing
+     * file, or a path that could be a regular output file.
+     */
+    public static boolean isRegularPath(final Path path) {
+        return !Files.exists(path) || Files.isRegularFile(path);
```

Comment:
This method is confusing to me, it seems like it should be named something else, but it mirrors what the existing method already does, so I guess we should leave it...